src/btree/bt_read.c

```diff
@@ -8,6 +8,333 @@
 
 #include "wt_internal.h"
 
+#define	WT_MAX_PREFIX_SIZE						\
+    (sizeof(char) +							\
+    sizeof(uint32_t) + sizeof(uint8_t) + WT_BTREE_MAX_ADDR_COOKIE)
+
+/*
+ * __las_build_prefix --
+ *	Build the unique file/address prefix.
+ */
+static void
+__las_build_prefix(WT_SESSION_IMPL *session, uint32_t btree_id,
+    const uint8_t *addr, size_t addr_size, uint8_t *prefix, size_t *prefix_lenp)
+{
+	size_t len;
+	void *p;
+
+	/*
+	 * Build the page's unique key prefix we'll search for in the lookaside
+	 * table, based on the file's ID and the page's block address.
+	 */
+	p = prefix;
+	*(char *)p = WT_LAS_RECONCILE_UPDATE;
+	p = (uint8_t *)p + sizeof(char);
+	memcpy(p, &btree_id, sizeof(uint32_t));
+	p = (uint8_t *)p + sizeof(uint32_t);
+	*(uint8_t *)p = (uint8_t)addr_size;
+	p = (uint8_t *)p + sizeof(uint8_t);
+	memcpy(p, addr, addr_size);
+	p = (uint8_t *)p + addr_size;
+	len = sizeof(char) + sizeof(uint32_t) + sizeof(uint8_t) + addr_size;
+	WT_ASSERT(session, WT_PTRDIFF(p, prefix) == len);
```

Comment:
@michaelcahill, @sueloverso:

It occurs to me we can't use the lookaside table for more than one purpose unless we make the key/value formats simplistic, that is, they both have to be WT_ITEMs that we parse explicitly, there's no reason to believe the key/value formats will be similar for other purposes.

Or is there some way around that problem?

I'd prefer not to crack the key/value pairs myself (and I'm confident Michael feels the same way).

If that's correct:

* Sue, I think we're going to have more internal tables in our future, we might want to move their file IDs somewhere out of the general file ID namespace.

* There's no reason to put a flag byte at the beginning of the record, I should get rid of WT_LAS_RECONCILE_UPDATE?

* We should rename the lookaside table because it can't ever be used for anything other than old reader information, and lookaside is a lousy name for that.

Thoughts?